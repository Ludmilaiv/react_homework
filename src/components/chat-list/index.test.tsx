import { ChatList } from './';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('ChatList', () => {
  it('render empty list', () => {
    render(<ChatList chats={[]} />);
  });

  it('render list', () => {
    const { asFragment } = render(<ChatList chats={[
      {id: '2', name: 'chat1'}, 
      {id: '1', name: 'chat2'}]} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render multiply components', () => {
    render(<>
      <ChatList chats={[]} />
      <ChatList chats={[]} />
    </>
    );

    expect(screen.queryAllByRole('list').length).toBe(2);
  });
});