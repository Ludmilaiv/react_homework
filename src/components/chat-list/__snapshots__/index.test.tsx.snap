// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatList render list 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <div
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          chat1
        </span>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <div
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          chat2
        </span>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
